core/corehttp/gateway_handler.go

```diff
@@ -174,15 +174,7 @@ func (i *gatewayHandler) getHandler(w http.ResponseWriter, r *http.Request) {
 
 	nd, p, err := i.ResolvePath(ctx, urlPath)
 	if err != nil {
-		if err == routing.ErrNotFound {
-			w.WriteHeader(http.StatusNotFound)
-		} else if err == context.DeadlineExceeded {
-			w.WriteHeader(http.StatusRequestTimeout)
-		} else {
-			w.WriteHeader(http.StatusBadRequest)
-		}
-
-		w.Write([]byte(err.Error()))
```

Comment:
erm, if you look at webError, it already handles everything this block of code did, but better, and also the ErrNoLink.  https://github.com/andyleap/go-ipfs/blob/gateway404/core/corehttp/gateway_handler.go#L451-L459 is almost exactly the same code.  It looks like someone was factoring out that method, but missed this code.  (note: this includes 408 for timeouts and such)